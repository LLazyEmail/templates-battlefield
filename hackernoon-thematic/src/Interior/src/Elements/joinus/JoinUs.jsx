import React from 'react';
import { styles } from './styles';
import { link } from './config';

import { Link } from 'react-emails-components-miscellaneous'; 

const JoinUs = () => (
  <div dir="ltr" style={styles.div}>
    <span style={styles.span1}>
      <span style={styles.span2}>
        <a href={link}
          style={styles.a}
          target="_blank">&quot;Join our writing contest and get a chance to win cool prizes&quot;.</a>
      </span>
    </span>
  </div>
);

export default JoinUs;