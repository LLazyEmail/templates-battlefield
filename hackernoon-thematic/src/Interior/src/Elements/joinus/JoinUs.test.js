import React from 'react';
import { create } from 'react-test-renderer';
import ReactDOM from 'react-dom';
import JoinUs from './index';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('JoinUs/index component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<JoinUs />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = create(<JoinUs />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Link has correct path', () => {
    const wrapper = shallow(<JoinUs />);
    const elem = wrapper.find('a');
    expect(elem.props()).toHaveProperty('href', 'https://hackernoon.com/the-decentralized-internet-writing-contest-fq1i35t4');
    expect(elem.props()).toHaveProperty('target', '_blank');
  });

  it('Link text check', () => {
    const wrapper = shallow(<JoinUs />);
    const elem = wrapper.find('a');
    expect(elem.render().text()).toEqual("\"Join our writing contest and get a chance to win cool prizes\".");
  });
});