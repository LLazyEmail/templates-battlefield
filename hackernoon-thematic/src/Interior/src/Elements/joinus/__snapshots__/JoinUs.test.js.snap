// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinUs/index component renders correctly 1`] = `
<div
  dir="ltr"
  style={
    Object {
      "textAlign": "justify",
    }
  }
>
  <span
    style={
      Object {
        "fontSize": "16px",
      }
    }
  >
    <span
      style={
        Object {
          "fontFamily": "trebuchet ms,lucida grande,lucida sans unicode,lucida sans,tahoma,sans-serif",
        }
      }
    >
      <a
        href="https://hackernoon.com/the-decentralized-internet-writing-contest-fq1i35t4"
        style={
          Object {
            "WebkitTextSizeAdjust": "100%",
            "color": "#111111",
            "fontWeight": "bold",
            "msTextSizeAdjust": "100%",
            "msoLineHeightRule": "exactly",
            "textDecoration": "underline",
          }
        }
        target="_blank"
      >
        "Join our writing contest and get a chance to win cool prizes".
      </a>
    </span>
  </span>
</div>
`;
